config.yml

```diff
@@ -42,11 +42,21 @@ defaults:
     apiVersion:       2014-01-01
 
   pulse:
-    username:         !env PULSE_USERNAME
-    password:         !env PULSE_PASSWORD
-    hostname:         !env PULSE_HOSTNAME
-    vhost:            !env PULSE_VHOST
-
+    namespace:                 !env PULSE_NAMESPACE
+    claimedCredentials:
+      namespace:               !env PULSE_NAMESPACE
+      expiresAfter:            !env PULSE_EXPIRES_AFTER
+      contact:                 !env PULSE_CONTACT
+      rootUrl:                 !env TASKCLUSTER_ROOT_URL
+      credentials:
+        clientId:              !env TASKCLUSTER_CLIENT_ID
+        accessToken:           !env TASKCLUSTER_ACCESS_TOKEN
+    client:
+      namespace:               !env PULSE_NAMESPACE
+      recycleInterval:         !env PULSE_CLIENT_RECYCLE_INTERVAL
+      retirementDelay:         !env PULSE_CLIENT_RETIREMENT_DELAY
+      minReconnectionInterval: !env PULSE_CLIENT_MIN_RECONNECTION_INTERVAL
```

Comment:
I think this is too much configuration :)

The only things that are really necessary here are the namespace, expiresAfter, and contact.  The client options can be left at their default values, and the others are available elsewhere in the config.  So the result would be

```js
pulse:
  namespace: !env PULSE_NAMESPACE
  contact: !env PULSE_CONTACT
  expiresAfter: !env PULSE_EXPIRES_AFTER
```

Then when constructing the client, use `cfg.taskcluster` for the credentials/rootUrl, and pass `cfg.pulse.namespace` as necessary.